util/session/sessionmanager.go

```diff
@@ -84,6 +85,10 @@ func (mgr *SessionManager) Create(subject string, secondsBeforeExpiry int64, id
 	// Create a new token object, specifying signing method and the claims
 	// you would like it to contain.
 	now := time.Now().UTC()
+	if id == "" && rbacpolicy.IsProjectSubject(subject) {
```

Comment:
It is not very intuitive why id is generated only for proj subject. I think we need to generate id in [project.CreateToken](https://github.com/argoproj/argo-cd/blob/master/server/project/project.go#L87)

Reply:
Moved id generating to project.CreateToken. Changed the logic in proj.GetJWTToken search by id(new token), if no id found then check by issued time(old token)